### ISSUE TITLE
Allow frame indexing for 6+ digits using the frame box

### ISSUE THREAD

**Issue by mfkeles** [Napari/napari#3795]
## 🧰 Task
I am using the napari-video plugin and I have some videos that are 1+ million frames. I noticed that the frames box does not let me enter more than 6 digits. Would it be possible to change this or give the user the ability to set a limit? I was able to fix this by following [this suggestion](https://forum.image.sc/t/frame-indexing-does-not-work-for-frame-id-6-digits/60947). As @Czaki pointed out the issue is related to the following line: https://github.com/napari/napari/blob/0582b41e9fffe81d0fe4a1af8fb961706b8619a7/napari/_qt/widgets/qt_dims_slider.py#L61

**Comment by jni** [comment]
Those are some big videos @mfkeles! At 24fps, 1M frames is 11.5h, or 4.5 at 60fps, or (perhaps more reasonable) 2h15min at 120fps! :astonished: 

Anyway, I agree we should just remove that limit — @tlambert03 @sofroniewn what do you think?

**Comment by tlambert03** [comment]
conceptually, absolutely in support of that.  in practice, the challenge with an unlimited spinbox is resizing the width appropriately.  in this case, though, I think we have an auto-resize already implemented there.  If that's the case, then there's no real issue with making it unbounded (but to guard against extreme/absurd cases, we should probably use an elided label with a max width)

**Comment by Czaki** [comment]
Maybe just update bounds when updating bounds of the slider?

**Comment by mfkeles** [comment]
> Those are some big videos @mfkeles! At 24fps, 1M frames is 11.5h, or 4.5 at 60fps, or (perhaps more reasonable) 2h15min at 120fps! 😲
> 
> Anyway, I agree we should just remove that limit — @tlambert03 @sofroniewn what do you think?

yep! it's a curse but I am working on the behavioral characterization of sleep stages, so usually 16-20 hour videos at 30fps!

**Comment by roaldarbol** [comment]
I'm running into the same problem (also with sleep staging by the way). I'd be happy to open a PR if that would be appreciated.

**Comment by TimMonko** [comment]
@roaldarbol, would be happy to see you open a PR! This seems like a reasonable thing to fix and if we don't have proper resizing machinery we should fix it. Let me/us know if you need any assistance 😁

**Comment by roaldarbol** [comment]
@TimMonko, Awesome - I'll have a go at it one of these days. I'll tag you in the PR for feedback. 😃
